cloud/scope/machine.go

```diff
@@ -328,10 +328,19 @@ func (m *MachineScope) ProviderID() string {
 
 // AvailabilitySet returns the availability set for this machine if available
 func (m *MachineScope) AvailabilitySet() (string, bool) {
-	if m.IsControlPlane() && m.AvailabilitySetEnabled() {
+	if len(m.FailureDomains()) > 0 {
+		return "", false
+	}
+
+	if m.IsControlPlane() {
 		return azure.GenerateAvailabilitySetName(m.ClusterName(), azure.ControlPlaneNodeGroup), true
 	}
 
+	// get machine deployment name from labels
+	if mdName, ok := m.Machine.Labels[clusterv1.MachineDeploymentLabelName]; ok {
```

Comment:
wondering if it makes more sense to use the MachineSet or the MachineDeployment name here 🤔  any particular reason you chose MD?

Reply:
Two reasons:
1. Better user experience since a cluster is mostly created with mds, I thought it'd be easier to map an md with an availability set.
2. I wasn't sure if there is a 1-1 mapping b/w md->ms. And it might end up with more than 1 as for the same md, again which I felt might be a bit odd.

Having said that, this will result in machines created with machine set will end up not being HA, so that's a concern. Maybe check for md, and then fallback to ms if not available?

Reply:
> Having said that, this will result in machines created with machine set will end up not being HA, so that's a concern. Maybe check for md, and then fallback to ms if not available?

I think that's okay. Let's keep it simple for now and stick with MachineDeployments only. Would you mind adding a line or two to https://capz.sigs.k8s.io/topics/failure-domains.html to describe the use of availability sets when failure domains are not available though?